### PR TITLE
added nicknames to envs

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
### What I did

- I added the `nickname` attribute to the `Env` and `NetworkEnv` objects. 
- Updated `pyproject.toml` to have `dev-dependencies`, so all dependencies can be in a single file, instead of split. 

If the dev-dependencies change is nice, I'm happy to update `CONTRIBUTING.md` documentation as such, and also the rest of the dependency management. 

### How I did it

```
self.nickname = "pyevm"
```

### How to verify it

```bash
pytest -k test_env_nickname
```

### Description for the changelog

- Added `nickname` attribute to `Env` and `NetworkEnv` classes

### Cute Animal Picture

image

### CODE REVIEW

File: boa/network.py [hunk 1]
```diff
@@ -201,6 +204,9 @@ def anchor(self):
             # wipe forked state
             self._reset_fork(block_number)
 
+    def set_nickname(self, nickname: str):
```
Comment (charles-cooper) [inline boa/network.py:207]: no need to override

File: boa/environment.py [hunk 2]
```diff
@@ -47,10 +47,15 @@ def __init__(self, fork_try_prefetch_state=False, fast_mode_enabled=False):
...
         self._gas_tracker = 0
 
+        self.nickname = "pyevm"
+
         self.evm = PyEVM(self, fast_mode_enabled, fork_try_prefetch_state)
 
     def set_random_seed(self, seed=None):
         self._random = random.Random(seed)
+    
+    def set_nickname(self, nickname):
```
Comment (charles-cooper) [inline boa/environment.py:57]: when would set_nickname be called?

**Review by charles-cooper — COMMENTED** [review]
please roll back the requirements / dev-dependencies changes, they are out of scope here

File: boa/environment.py (hunk 2 above)
Comment (PatrickAlphaC) [reply boa/environment.py:57]: Maybe you have 2 ZKSync branches and you want to call one `zksync-testnet` and one `zksync-mainnet`

Comment (charles-cooper) [reply boa/environment.py:57]: yea but it is already set at construction, so i'm not sure why set_nickname would be called after construction

Comment (PatrickAlphaC) [reply boa/environment.py:57]: If you’re in a Python shell and you setup zksync and called it zksync, but then realized you have 2 zksync chains, you’d have to just make two new ones with new names? Feels awkward to then have 3.

- zksync 
- Zksync testnet 
- Zksync mainnet

File: boa/network.py [hunk 3]
```diff
@@ -164,6 +165,8 @@ def __init__(
 
         self._rpc: RPC = rpc
 
+        self.nickname = nickname or rpc.identifier
```
Comment (charles-cooper) [inline boa/network.py:168]: `rpc.name` is more likely to strip sensitive info out of the url
```suggestion
        self.nickname = nickname or rpc.name
```

**Pull request merged**